### PR TITLE
Gdal: Fix conflicts

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **gdal/***

It's just not possible to build the current gdal recipe, cf https://github.com/eirikb/proof-of-conan/actions/runs/8718677788
the conflicts are:
- `libgeotiff/1.7.1->libtiff/4.6.0, gdal/3.4.3->libtiff/4.5.1.` 
- `libtiff/4.6.0->libdeflate/1.19, gdal/3.4.3->libdeflate/1.18.`
- `libgeotiff/1.7.1->proj/9.3.1, gdal/3.4.3->proj/9.2.1.`
- `proj/9.3.1->sqlite3/3.44.2, gdal/3.4.3->sqlite3/3.42.0.`

also, [libpq/14.5](https://github.com/conan-io/conan-center-index/tree/HEAD/recipes/libpq/config.yml) and [openssl/1.1.1u](https://github.com/conan-io/conan-center-index/tree/HEAD/recipes/openssl/config.yml) are not maintained on CCI any more

The other requirements modification are the transition to ranges for zlib, libpng and libcurl

Also, drop dead code, and 3.4.1 and 3.3.3 which are not used

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/7446063795/attempts/1)

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/7446063795/attempts/2)

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/7446063795/attempts/3)

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/7446063795/attempts/5)

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/7446063795/attempts/6)

**Review by jwillikers — APPROVED** [review]

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/8552364239/attempts/1)

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/8552364239/attempts/2)

**Review by valgur — APPROVED** [review]

**Comment by uilianries** [comment]
@ericLemanissier Thank you for your PR. Could you please update your PR description with what conflicts you are trying to solve? Do you have an associated PR/issue that can be linked?

**Comment by ericLemanissier** [comment]
It's just not possible to build the current gdal recipe, cf https://github.com/eirikb/proof-of-conan/actions/runs/8718677788
the first conflict is `libgeotiff/1.7.1->libtiff/4.6.0, gdal/3.4.3->libtiff/4.5.1.`

File: recipes/gdal/pre_3.5.0/conanfile.py [hunk 1]
```diff
@@ -434,8 +386,7 @@ def _patch_sources(self):
             # os.path.join("ogr", "ogrsf_frmts", "cad", "libopencad"), # TODO: uncomment when libopencad available
             os.path.join("ogr", "ogrsf_frmts", "geojson", "libjson"),
         ]
-        if Version(self.version) >= "3.1.0":
-            embedded_libs.append(os.path.join("ogr", "ogrsf_frmts", "flatgeobuf", "flatbuffers"))
+        embedded_libs.append(os.path.join("ogr", "ogrsf_frmts", "flatgeobuf", "flatbuffers"))
```
Comment (Ahajha) [inline recipes/gdal/pre_3.5.0/conanfile.py:389]: embedded_libs is declared as a list a few lines above, can probably move this new item directly into the declaration.

File: recipes/gdal/pre_3.5.0/conanfile.py [hunk 2]
```diff
@@ -205,20 +174,6 @@ def config_options(self):
             del self.options.fPIC
         # if Version(self.version) < "3.0.0":
         #     del self.options.with_tiledb
```
Comment (Ahajha) [inline recipes/gdal/pre_3.5.0/conanfile.py:176]: These can probably be removed as well

File: recipes/gdal/pre_3.5.0/conanfile.py (hunk 1 above)
Comment (ericLemanissier) [reply recipes/gdal/pre_3.5.0/conanfile.py]: thanks !

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/8552364239/attempts/3)

**Review by jwillikers — APPROVED** [review]

**Comment by ericLemanissier** [comment]
@uilianries it's ready for review

**Comment by ericLemanissier** [comment]
@uilianries @RubenRBS I simplified this PR by not migrating to f-strings. Is it possible to get a review ?

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/8552364239/attempts/4)

**Comment by ericLemanissier** [comment]
@jcar87 how can we make progress on this one ? is the description clear enough ? is the change ok ?

**Review by jwillikers — APPROVED** [review]

**Comment by uilianries** [comment]
The pdal package uses the removed version 3.4.1: https://github.com/conan-io/conan-center-index/blob/master/recipes/pdal/all/conanfile.py#L74

**Comment by ericLemanissier** [comment]
Yes, bus as it's conan 1 only, and not touched in the last 2 years, it'll need an important refactoring anyway

**Comment by ericLemanissier** [comment]
@uilianries I restored 3.4.1

**Review by uilianries — COMMENTED** [review]
Looking good, only a small comment.

File: recipes/gdal/pre_3.5.0/conanfile.py [hunk 3]
```diff
@@ -523,6 +468,12 @@ def replace_in_nmake_opt(str1, str2):
             replace_in_file(self, os.path.join(self.source_folder, "alg", "makefile.vc"),
                                   "$(GEOS_CFLAGS)", "$(GEOS_CFLAGS) /DHAVE_OPENCL")
 
+        # Workaround for a NMakeDeps define quoting issue:
+        # https://github.com/conan-io/conan/issues/13603
+        replace_in_file(self, os.path.join(self.generators_folder, "conannmakedeps.bat"),
+                        r'/DSQLITE_API#\"__declspec(dllimport)\"',
+                        "/DSQLITE_API#__declspec(dllimport)", strict=False)
+
```
[the remark this answers is not in the document]
Comment (ericLemanissier) [reply recipes/gdal/pre_3.5.0/conanfile.py:476]: yet it failed on January 10th https://github.com/conan-io/conan-center-index/issues/23660

[the remark this answers is not in the document]
Comment (uilianries) [reply recipes/gdal/pre_3.5.0/conanfile.py]: Yes, but it was in January. Right now we are running 1.64.0 and 2.2.2:

- https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6
- https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v2.yml#L6

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/9028694774/attempts/1)

**Review by uilianries — APPROVED** [review]
LGTM

**Review by valgur — APPROVED** [review]

**Comment by ericLemanissier** [comment]
Thank yo for the approvals, it's been a long one !

**Pull request merged**